src/index.js

```diff
@@ -317,7 +317,17 @@ async function traverse(directory) {
         await traverse(itemPath);
       } else if (stat.isFile()) {
         let extension = path.extname(itemPath).toLowerCase();
-        if (extension === ".js" || extension === ".cjs" || extension === "") {
+        let isCommonJS = false;
+        if (
+          _app.type === "commonjs" &&
+          (extension === ".js" || extension === ".cjs" || extension === "")
+        ) {
+          isCommonJS = true;
+        } else if (_app.type === "module" && extension === ".cjs") {
+          isCommonJS = true;
```

Comment:
probably, here isCommonJS must be false

Reply:
Friday Evan and I discussed about this commit. I made a flow chart Friday and we are trying to revise/write the code for this flow chart. I will email you the flow chart.